### DISCUSSION TITLE
torrent Drop during torrent.MergeSpec

### DISCUSSION THREAD

**Discussion by ghost** [anacrolix/torrent#511] [General]
there is chance that torrent maybe dropped during torrent.MergeSpec.

is it mutex protected?

**Comment by anacrolix** [comment]
I believe it is safe. The parts of MergeSpec that occur without the lock, are to the public methods that should themselves be safe. I would not provide any guarantees about whether it is valid to race Torrent.Drop, and Torrent.MergeSpec, but if you spot anything feel free to submit a fix.
